### ISSUE TITLE
(3.0.50) debugger frame trimmed too far on `halt`

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#330]
I sent the message `halt` so that I could inspect the receiver of the `halt` message but the stack has been trimmed to the point where the frame with the receiver is not visible in debugger:
 ![image](https://user-images.githubusercontent.com/623951/49049479-fa805100-f193-11e8-9986-5d850c9684d1.png)
It _is_ convenient to position the debugger at the frame that sends `halt` since that is often a good default, however, trimming the stack removes useful frames from the stack that prvide information that is otherwise unreachable ...

**Comment by LisaAlmarode** [comment]
The method frame in which the halt is sent should not be the top of the debugger.  
In GBS/VW, there are three further frames which may be overkill, but at least the Object >> halt frame ought to be there.
GBS/VW selects the top frame so you do have to select the frame you want but this isn't a big deal.

**Comment by rjsargent** [comment]
VA Smalltalk does a good of selecting the stack frame in which the exception was signalled. I have found this to be a useful default for the debugger. That would correspond to selecting the Object>>#halt frame.
(Although, I think VA Smalltalk recognizes #halt and #halt: as special cases and chooses their senders first.)

**Comment by ericwinger** [comment]
Here are a list of stack frames which the JadeiteDebugger will trim: 
```
	'(AbstractException) >> _signalAsync @7 line 21'
		'(Object) >> _doesNotUnderstand:args:envId:reason: @7 line 12'
		'(Object) >> _doesNotUnderstand:args:envId:reason: @8 line 13'
		'(Object) >> error: @6 line 7'
		'(Object) >> error: (envId 0) @6 line 7'
		'(Object) >> error: @6 line 7'
		'(Object) >> halt @2 line 5'
		'(Object) >> halt (envId 0) @2 line 5'
		'(Object) >> pause @2 line 7'
		'(Object)>>#_doesNotUnderstand: @1 line 6'
		'(Object)>>#error: @1 line 6'
		'(Object)>>#error: @3 line 7'
		'(Object)>>#halt @1 line 6'
		'(Object)>>#pause @2 line 8'
		'(TestCase)>>#assert: @1 line 1'
		'(TestCase)>>#deny: @2 line 3'
		'(TestCase)>>#should:raise: @10 line 9'
		'(TestCase)>>#shouldnt:raise: @4 line 2'
		'GSTestCase >> assert: @1 line 1'
		'GSTestCase >> assert: (envId 0) @1 line 1'
		'Object >> _doesNotUnderstand:'
		'Object >> _doesNotUnderstand:args:envId:reason: @7 line 12'
		'Object >> _doesNotUnderstand:args:envId:reason: (envId 0) @7 line 12'
		'Object >> _doesNotUnderstand:args:envId:reason: @8 line 13'
		'Object >> error: @6 line 7'
		'Object >> error: (envId 0) @6 line 7'
		'(TestAsserter) >> assert: @5 line 4'
		'TestAsserter >> assert: (envId 0) @5 line 4'
```

Although we could do more selective trimming the easiest would be to not trim at all and select the former top of stack as the selection frame. Here are three examples of how the debugger would open. Would this satisfy the demand? @rjsargent @LisaAlmarode @dalehenrich 

![image](https://user-images.githubusercontent.com/36212516/50234955-a9671580-036b-11e9-8515-26ba9a06215f.png)

![image](https://user-images.githubusercontent.com/36212516/50235118-019e1780-036c-11e9-8dbd-a96623b811f8.png)


![image](https://user-images.githubusercontent.com/36212516/50234990-bf74d600-036b-11e9-966e-ed11553a5394.png)

**Comment by ericwinger** [comment]
Assigning to @LisaAlmarode for feedback

**Comment by ericwinger** [comment]
Have a fix that seems solid in Oscar 3.0.54. Stack no longer trimmed, but selected frame is the frame that raised the error (i.e. - right before the halt frame)
